### ISSUE TITLE
tensorboard monitoring

### ISSUE THREAD

**Issue by Joshuaalbert** [Joshuaalbert/neural_deprojection#34]
TB functionality is added.
See example here: https://github.com/Joshuaalbert/neural_deprojection/blob/master/neural_deprojection/models/identify_medium/main.py#L167

You need to use the config object I've defined:

```python
config = dict(model_type='model1',
                  model_parameters=dict(num_layers=3, image_feature_size=4),
                  optimizer_parameters=dict(learning_rate=1e-5, opt_type='adam'),
                  loss_parameters=dict())
```

You can add/remove any items in the lower level dicts, but you need to keep the upper dict structure the same. Only add config items that you actually use, i.e. don't include `num_layers` if you're not using it to define your model. 

Note, these lines to build the checkpoint and log dirs: https://github.com/Joshuaalbert/neural_deprojection/blob/master/neural_deprojection/models/identify_medium/main.py#L183-L184

 Use something like these lines to plot things like your image:
https://github.com/Joshuaalbert/neural_deprojection/blob/master/neural_deprojection/models/identify_medium/main.py#L93-L94

I've already added loss, and distribution of the weights.

To see your results you need to run this on hpc1 (with conda sourced of course):

tensorboard --logdir BASE_LOG_DIR --port 6666

where BASE_LOG_DIR is whatever you chose in the earlier steps.

Then on your laptop run this:

```
ssh -L 6666:localhost:6666 hpc1
```

Then go to localhost:6666 on your laptop.

Examples:

![Screenshot from 2021-03-16 13-57-45](https://user-images.githubusercontent.com/14807032/111357371-89794600-8689-11eb-9371-5fbd22dc6822.png)
![Screenshot from 2021-03-16 13-57-16](https://user-images.githubusercontent.com/14807032/111357378-8a11dc80-8689-11eb-9ca2-127336cdd352.png)

**Comment by Joshuaalbert** [comment]
Then once you have done some training and it's all logged, then click on upload:
![Screenshot from 2021-03-16 15-18-53](https://user-images.githubusercontent.com/14807032/111367511-1bd31700-8695-11eb-86b1-7467357f6a96.png)

It'll give you a command to run in your terminal to make a publically visible project:
E.g. here my run on random data.

https://tensorboard.dev/experiment/VgG8VtVaSyCQHxKoIXDkWQ
